src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -79,7 +79,10 @@ async def exec_line_async(
         async with httpx.AsyncClient() as client:
             response = await client.post(url, json=payload, timeout=LINE_TIMEOUT_SEC)
         response = self.process_http_response(response)
-        return LineResult.deserialize(response)
+        line_result = LineResult.deserialize(response)
+        # TODO: move this logic to the executor service?
+        line_result.run_info.inputs[LINE_NUMBER_KEY] = index
```

Comment:
why python code works without adding this?

Reply:
updated inline comment
# local_storage_operations.load_inputs_and_outputs now have an assumption that there is an extra
        # line_number key in the inputs.
        # This key will be appended to the inputs in below call stack:
        # BatchEngine.run =>
        # BatchInputsProcessor.process_batch_inputs =>
        # ... =>
        # BatchInputsProcessor._merge_input_dicts_by_line
        # For python, it will be kept in the returned line_result.run_info.inputs
        # For csharp, it will be dropped by executor service for now
        # Append it here for now to make behavior consistent among ExecutorProxy.